### ISSUE TITLE
invalid_grant

### ISSUE THREAD

**Issue by SoftDevDanial** [areed1192/td-ameritrade-python-api#131]
Hi,

While i was testing out this library, i used my account to test the features and it was working fine(just the basic get quotes only). Then i wanted to get_transactions but on my dads account number which obviously will get permission error. So i wanted to change the username and password but didn't know how from the code itself since it kept using the cache refresh token(probably is logout i guess which i didnt know at first) . So i went and changed the username and password at the auth url. After changing it, i still get the permission error. So i tried deleting the cache and deleting the app and creating a new one and also going to the auth url and log in. Now im getting this error below which im not sure how to handle. If i do it via https://developer.tdameritrade.com/authentication/apis/post/token-0 its a success but if i use the code its not. 
> td.exceptions.NotNulError: {
  "error" : "invalid_grant"
}

I assume its the refresh token thats causing this.

**Comment by areed1192** [comment]
Yeah, when you see an invalid grant, which means the refresh token you provided was incorrect. What you have to do is delete the JSON file you specify when you create a new instance of the client. That should fix the issue.

**Comment by areed1192** [comment]
this has been fixed, please update the library to version 0.3.2

**Comment by SoftDevDanial** [comment]
Hi, 
What do i put in the credentials file(syntax etc) and what file type? Quite new in this.

**Comment by areed1192** [comment]
@SoftDevDanial You don't need to put anything in the credential file it will be auto-populated for you. All you need to provide is a path where you want to store the credentials. For example,

`C:\Users\Alex\OneDrive\Desktop\Credentials\td_state.json`

**Comment by SoftDevDanial** [comment]
@areed1192 
Hi , im getting this error after putting in the json file. This is my code.
```
#Import the client
from td.client import TDClient


# Create a new session, credentials path is optional.
TDSession = TDClient(
    client_id='**',
    redirect_uri='http://localhost',
    credentials_path=r"**\credentials.json",
)

# Login to the session
TDSession.login()
```

**Comment by areed1192** [comment]
That error is saying that the JSON file is empty. Did you validate that something is in the JSON file?

**Comment by SoftDevDanial** [comment]
my json file is blank.. is it probably because of that? Do i need to add like
`
USERNAME:
ACCOUNT:
`
etc?

**Comment by areed1192** [comment]
Nope, all you need to do is simply provide the full path to where you want the tokens to be cached. Just take the code you have up above and provide the file path.:

```python
from td.client import TDClient

# Create a new session, credentials path is optional.
TDSession = TDClient(
    client_id='**',
    redirect_uri='http://localhost',
    credentials_path=r"<FULL_FILE_PATH>"
)

# Login to the session
TDSession.login()
```

You'll be prompted to redirect to the login URL, follow that process and then take the authorization code and paste it back into the terminal. For more detailed instructions refer to the README.md. It has all the steps and pictures to help guide you.

**Comment by SoftDevDanial** [comment]
Yep already did and i the json file is blank. Code is the same as yours.

**Comment by areed1192** [comment]
Well, it has something to do with the account credentials you're passing through. Your Dad's account is registered with the TD API, right?

**Comment by pracplayopen** [comment]
After doing this many times w/out incident, today I started experiencing this error again on a new tda account.

None of standard fixes for invalid_grant were working.

Then I realized that the client-key is also provided in the url used to login and request the one-time use code prior to requesting a token.

So just make sure everything matches everywhere, including in the url for requesting the one-time code.
